File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN ; int sum =0 ;

        for(auto i : nums){
            sum += i ;
            if(sum > ans){
                ans = sum ;
            }
            if(sum<0){
                sum = 0 ;
            }
        }
        return ans ;
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0 
        ans = -10000000

        for i in nums:
            sum += i
            if(sum > ans):
                ans = sum
            if(sum<0):
                sum = 0
        
        return ans